src/Cli.Tests/AddTelemetryTests.cs

```diff
@@ -0,0 +1,166 @@
+                Assert.AreEqual(isTelemetryEnabled is CliBool.True ? true : false, telemetryOptions.ApplicationInsights.Enabled);
+                Assert.AreEqual(appInsightsConnString, telemetryOptions.ApplicationInsights.ConnectionString);
+            }
+        }
+
+        /// <summary>
+        /// Test to verify when Telemetry section is present in the config
+        /// It should add application insights telemetry if telemetry section is empty
+        /// or overwrite the existing app insights telemetry with the given app insights telemetry options.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Add AppInsights Telemetry when telemetry section is empty.")]
+        [DataRow(false, DisplayName = "Overwrite AppInsights Telemetry when telemetry section already exists.")]
+        public void TestAddAppInsightsTelemetryWhenTelemetryAlreadyExists(bool isEmptyTelemetry)
+        {
+            MockFileSystem fileSystem = FileSystemUtils.ProvisionMockFileSystem();
+            string configPath = "test-app-insights-config.json";
+            string runtimeSection = isEmptyTelemetry ? RUNTIME_SECTION_WITH_EMPTY_TELEMETRY_SECTION : RUNTIME_SECTION_WITH_APP_INSIGHTS_TELEMETRY_SECTION;
+            string configData = $"{{{SAMPLE_SCHEMA_DATA_SOURCE},{runtimeSection}}}";
+            fileSystem.AddFile(configPath, new MockFileData(configData));
+
+            // Initial State
+            Assert.IsTrue(fileSystem.FileExists(configPath));
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(fileSystem.File.ReadAllText(configPath), out RuntimeConfig? config));
+            Assert.IsNotNull(config);
+            Assert.IsNotNull(config.Runtime);
+            Assert.IsNotNull(config.Runtime.Telemetry);
+
+            if (isEmptyTelemetry)
+            {
+                Assert.IsNull(config.Runtime!.Telemetry!.ApplicationInsights);
+            }
+            else
+            {
+                Assert.IsNotNull(config.Runtime!.Telemetry!.ApplicationInsights);
+                Assert.AreEqual(true, config.Runtime!.Telemetry!.ApplicationInsights!.Enabled);
+                Assert.AreEqual("InstrumentationKey=00000000-0000-0000-0000-000000000000", config.Runtime!.Telemetry!.ApplicationInsights!.ConnectionString);
+            }
+
+            // Add Telemetry
+            bool isSuccess = ConfigGenerator.TryAddTelemetry(
+                new AddTelemetryOptions("InstrumentationKey=11111-1111-111-11-1", CliBool.False, configPath),
+                new FileSystemRuntimeConfigLoader(fileSystem),
+                fileSystem);
+
+            // Assert after adding telemetry
+            Assert.IsTrue(isSuccess);
+            Assert.IsTrue(fileSystem.FileExists(configPath));
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(fileSystem.File.ReadAllText(configPath), out config));
+            Assert.IsNotNull(config);
+            Assert.IsNotNull(config.Runtime!.Telemetry);
```

Comment:
nit: Same suggestion ~ asserting `config.Runtime` to be not null could help get away with `!` operator